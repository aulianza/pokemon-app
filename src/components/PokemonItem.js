import React, { useState } from "react";
import { useHistory } from "react-router-dom";
import { ColorExtractor } from "react-color-extractor";
import Button from "../components/Button";
import Card from "../components/Card";
import "./PokemonItems.css";

const PokemonItem = (props) => {
    const [colorState, setColorState] = useState("");

    let route = useHistory();
    const clickHandler = () => {
        route.push(`/pokemon/${props.poke.name}`);
    };

    const releaseHandler = () => {
        props.onRelease(props.poke.nickname);
    };

    const filterColor = (color) => {
        setColorState(color[1]);
    };

    return (
        <>
            <ColorExtractor src={props.poke.image} getColors={filterColor} />
            <Card
                className="pokemon-list__item"
                inlineStyle={{ backgroundColor: colorState }}
                isCardWave
            >
                <div onClick={clickHandler}>
                    <div className="pokemon-list__name">{props.poke.name}</div>
                    <div className="pokemon-list__image">
                        <img
                            src={props.poke.image}
                            alt={props.poke.name}
                            width="150"
                            height="150"
                        />
                    </div>
                    {props.currentPage === "pokemonList" ? (
                        <div className="pokemon-list__description">
                            <span>Owned : {props.poke.owned}</span>
                        </div>
                    ) : (
                        <div
                            className="pokemon-list__description"
                            style={{ paddingTop: "0" }}
                        >
                            <span>Nickname :</span>
                            <span className="pokemon-list__nickname">
                                {props.poke.nickname}
                            </span>
                        </div>
                    )}
                </div>
                {props.releaseAction === true ? (
                    <div className="pokemon-list__action">
                        <Button
                            onButtonClicked={releaseHandler}
                            title="Release"
                            buttonColor="btn-danger"
                            isBlock="true"
                        />
                    </div>
                ) : (
                    ""
                )}
            </Card>
        </>
    );
};

export default PokemonItem;
