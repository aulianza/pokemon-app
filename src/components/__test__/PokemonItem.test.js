import PokemonItem from "../PokemonItem";
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from "react-dom/test-utils";

let container = null;

beforeEach(() => {
    container = document.createElement("div");
    document.body.appendChild(container);
});

afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
});

it("pokemon-item-component, snapshot", () => {
    act(() => {
        render(<PokemonItem />, container)
    })
    expect(container).toMatchSnapshot();
});
