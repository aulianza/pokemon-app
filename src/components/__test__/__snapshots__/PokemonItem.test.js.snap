// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokemon-item-component, snapshot 1`] = `
<div>
  <div
    class="css-ywa3ly"
    width="32%"
  >
    <div
      class="css-1nvs4wv"
    >
      <div>
        <div
          class="pokemon-list__name"
        />
        <div
          class="pokemon-list__image"
        >
          <img
            height="150"
            width="150"
          />
        </div>
        <div
          class="pokemon-list__description"
          style="padding-top: 0px;"
        >
          <span>
            Nickname :
          </span>
          <span
            class="pokemon-list__nickname"
          />
        </div>
      </div>
      
    </div>
  </div>
</div>
`;
